id-to-url.js

```diff
@@ -1,14 +1,37 @@
 const DEFAULT_PORT = require('./port');
 
+const base = `http://localhost:${DEFAULT_PORT}`
+
 module.exports = function idToUrl(blobId, params) {
-  const port = (params && params.port) || DEFAULT_PORT;
+  const url = new URL(base)
+
   const [pureBlobId, query] = blobId.split('?');
-  const blobRef = encodeURIComponent(pureBlobId);
-  const paramsStr = query
-    ? '?' + query
-    : params && params.unbox
-    ? `?unbox=${encodeURIComponent(params.unbox.toString('base64'))}.boxs`
-    : '';
-
-  return `http://localhost:${port}/get/${blobRef}${paramsStr}`;
+
+  if (params && params.hostname) url.hostname = params.hostname;
+  if (params && params.port) url.port = params.port;
+
+  url.pathname = `/get/${encodeURIComponent(pureBlobId)}`;
+
+  const unbox = extractKey(query) || (params && params.unbox && toString(params.unbox));
+  if (unbox) url.searchParams.set('unbox', unbox + '.boxs');
+
+  return url.href
 };
+
+function extractKey (query) {
+  if (!query) return
+
+  return (
+    query.startsWith('unbox=') &&
+    query
+      .replace('unbox=', '')
+      .replace('.boxs', '')
+  )
+}
+
+function toString (key) {
+  if (typeof key === 'string') return key
+  if (Buffer.isBuffer) return key.toString('base64')
```

Comment:
```suggestion
  if (Buffer.isBuffer(key)) return key.toString('base64')
```

Important.

Reply:
Can you provide a test case for `params.unbox` being a Buffer? Or then, just remove this subfeature and assume it's a string always?

Reply:
I ran the tests as is and a buffer came through from somewhere. YAY tests
i.e. this was TDD, and it wasn't my test